ICAFIX/scripts/functionhighpassandvariancenormalize.m

```diff
@@ -129,26 +137,42 @@ function functionhighpassandvariancenormalize(TR,hp,fmri,WBC,varargin)
 
 elseif hp>0  % "fslmaths -bptf" based filtering
     if dovol > 0
+        % Save and filter confounds, as a NIFTI, as expected by FIX
         save_avw(reshape(confounds',size(confounds,2),1,1,size(confounds,1)),[fmri hpstring '.ica/mc/prefiltered_func_data_mcf_conf'],'f',[1 1 1 TR]);
         call_fsl(sprintf(['fslmaths ' fmri hpstring '.ica/mc/prefiltered_func_data_mcf_conf -bptf %f -1 ' fmri hpstring '.ica/mc/prefiltered_func_data_mcf_conf_hp'],0.5*hp/TR));
 
-        save_avw(reshape(cts,ctsX,ctsY,ctsZ,ctsT),[fmri hpstring '.nii.gz'],'f',[1 1 1 TR]);
-        call_fsl(['fslmaths ' fmri hpstring '.nii.gz -bptf ' num2str(0.5*hp/TR) ' -1 ' fmri hpstring '.nii.gz']);
-        cts=single(read_avw([fmri hpstring '.nii.gz']));
-        cts=reshape(cts,ctsX*ctsY*ctsZ,ctsT);
-        call_fsl(['fslcpgeom ' fmri '.nii.gz ' fmri hpstring '.nii.gz -d']);
+        % bptf filtering; no masking here, so this is probably memory inefficient
+        call_fsl(['fslmaths ' fmri '.nii.gz -bptf ' num2str(0.5*hp/TR) ' -1 ' fmri hpstring '.nii.gz']);
```

Comment:
Maybe add a comment here that it is restarting from the input file, rather than using the contents of cts, to make it more obvious what is going on.

It should probably also clear cts before the filtering command - if you are ambitious, you could move the first cts loading into all other conditions of this conditional of this conditional (and add the code to generate the mask here) so it doesn't waste time loading it twice.

Reply:
Never mind, I see that the loading section at the top was in fact deleted.